backend/asmlink.ml

```diff
@@ -85,19 +85,18 @@ let check_consistency file_name unit crc =
     raise(Error(Inconsistent_implementation(name, user, auth)))
   end;
   begin try
-    let source = List.assoc unit.ui_name !implementations_defined in
+    let source = String.Tbl.find implementations_defined unit.ui_name in
     raise (Error(Multiple_definition(unit.ui_name, file_name, source)))
   with Not_found -> ()
   end;
   implementations := unit.ui_name :: !implementations;
   Cmx_consistbl.set crc_implementations unit.ui_name crc file_name;
-  implementations_defined :=
-    (unit.ui_name, file_name) :: !implementations_defined;
+  String.Tbl.add implementations_defined unit.ui_name file_name;
   if unit.ui_symbol <> unit.ui_name then
     cmx_required := unit.ui_name :: !cmx_required
 
 let extract_crc_interfaces () =
-  Cmi_consistbl.extract !interfaces crc_interfaces
+  Cmi_consistbl.extract (String.Set.elements !interfaces) crc_interfaces
```

Comment:
Seems a shame to convert the set to a list just to unnecessarily sort it and fold over it. I think you should add an `extract_from_set` (or `extract_from_table`) function to `Consistbl`, and probably rename `extract_map` to `extract_map_from_set` for consistency.

Reply:
Yeah, ok.  Although this is only used in the packing code, I think.

Reply:
Actually it's easy just to use the new `find` function in `Consistbl` here.